comet/scrapers/bitmagnet.py

```
@@ -0,0 +1,101 @@
+        except Exception as e:
+            logger.warning(f"Error parsing torrent item from BitMagnet: {e}")
+            continue
+    return torrents
+
+
+async def scrape_bitmagnet_page(session, url, query, offset, limit):
+    try:
+        params = {"t": "search", "q": query, "offset": offset, "limit": limit}
+        async with session.get(f"{url}/torznab/api", params=params) as response:
+            data_text = await response.text()
+            root = ET.fromstring(data_text)
```

Ruff (0.14.5)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)